src/main/java/com/celements/webform/DocFormScriptService.java

```diff
@@ -83,6 +101,24 @@ private XWikiContext getContext() {
     }
     return docs;
   }
+
+  public Map<String, Set<DocumentReference>> updateAndSaveDocFromRequest() {
+    return updateAndSaveDocFromRequest(null);
+  }
+
+  @SuppressWarnings("unchecked")
+  public Map<String, Set<DocumentReference>> updateAndSaveDocFromRequest(
+      DocumentReference docRef) {
+    Collection<XWikiDocument> xdocs = Collections.<XWikiDocument>emptyList();
+    try {
+      xdocs = getDocFormCommand().updateDocFromMap(docRef,
+          getContext().getRequest().getParameterMap(), getContext());
+      return saveXWikiDocCollection(xdocs);
```

Comment:
Add rights checks at this point and only call saveXWikiDocCollection on docFormCommand if the rights are given. We need the rights-checks here because it is a call comming from scripts. In an other situation we can have a direct call out of a java process (even without a request, but only a value/key map) where we do not want to additionally check for access rights.